src/streams/EditStreamCard.js

```diff
@@ -5,45 +5,96 @@ import { View } from 'react-native';
+    is_web_public: boolean | void,
+    history_public_to_subscribers: boolean,
   |},
-  onComplete: (name: string, description: string, isPrivate: boolean) => void,
+  onComplete: (name: string, description: string, policySettings: StreamPolicySettings) => void,
 |}>;
 
 type State = {|
   name: string,
   description: string,
-  isPrivate: boolean,
+  accessPolicy: AccessPolicy,
 |};
 
+const accessPolicyFromStreamSettings = (streamSettings: StreamPolicySettings): AccessPolicy => {
+  const { isWebPublic, isPrivate, historyPublicToSubscribers } = streamSettings;
+  if (isWebPublic) {
+    return 'web_public';
+  } else if (!isPrivate) {
+    return 'public';
+  } else if (isPrivate && historyPublicToSubscribers) {
+    return 'private_shared_history';
+  } else {
+    return 'private_protected_history';
+  }
+};
+
+const streamAccessPolicies = {
+  web_public: {
+    title: 'Web-public',
+    policy: { isPrivate: false, isWebPublic: true, historyPublicToSubscribers: true },
+    description:
+      'Organization members can join (guests must be invited by a subscriber); anyone on the Internet can view complete message history without creating an account',
+  },
+  public: {
+    title: 'Public',
+    policy: { isPrivate: false, isWebPublic: false, historyPublicToSubscribers: false },
+    description:
+      'Organization members can join (guests must be invited by a subscriber); organization members can view complete message history without joining',
+  },
+  private_shared_history: {
+    title: 'Private, shared history',
+    policy: { isPrivate: true, isWebPublic: false, historyPublicToSubscribers: true },
+    description:
+      'Must be invited by a subscriber; new subscribers can view complete message history; hidden from non-administrator users',
+  },
+  private_protected_history: {
+    title: 'Private, protected history',
+    policy: { isPrivate: true, isWebPublic: false, historyPublicToSubscribers: false },
+    description:
+      'Must be invited by a subscriber; new subscribers can only see messages sent after they join; hidden from non-administrator users',
+  },
+};
+
 export default class EditStreamCard extends PureComponent<Props, State> {
   state: State = {
     name: this.props.initialValues.name,
     description: this.props.initialValues.description,
-    isPrivate: this.props.initialValues.invite_only,
+    accessPolicy: accessPolicyFromStreamSettings({
+      isPrivate: this.props.initialValues.invite_only,
+      isWebPublic: this.props.initialValues.is_web_public ?? false,
+      historyPublicToSubscribers: this.props.initialValues.history_public_to_subscribers,
+    }),
   };
 
   handlePerformAction: () => void = () => {
     const { onComplete } = this.props;
-    const { name, description, isPrivate } = this.state;
-    onComplete(name, description, isPrivate);
+    const { name, description, accessPolicy } = this.state;
+    onComplete(name, description, streamAccessPolicies[accessPolicy].policy);
```

Comment:
Using the map here is simple but at a glance it looks like a opportunity for a keyerror. Question: is flow checking the possible values here and ensuring that there will not be a lookup error?

Reply:
Ultimately, when I'm not sure of the answer to a question like that, I like to test it empirically.

Here, try munging one of the names somewhere as if there were a typo -- for example like:
```diff
 const streamAccessPolicies = {
-  web_public: {
+  webb_public: {
     title: 'Web-public',
```

That causes an error at this property access. The full error message is:
```
Error ------------------------------------------------------------------------------ src/streams/EditStreamCard.js:97:56

Cannot get `streamAccessPolicies[accessPolicy]` because property `web_public` (did you mean `webb_public`?) is missing
in object literal [1]. [prop-missing]

   src/streams/EditStreamCard.js:97:56
   97|     onComplete(name, description, streamAccessPolicies[accessPolicy].policy);
                                                              ^^^^^^^^^^^^

References:
   src/streams/EditStreamCard.js:56:30
                                    v
   56| const streamAccessPolicies = {
   57|   webb_public: {
   58|     title: 'Web-public',
   59|     policy: { isPrivate: false, isWebPublic: true, historyPublicToSubscribers: true },
   60|     description:
   61|       'Organization members can join (guests must be invited by a subscriber); anyone on the Internet can view complete message history without creating an account',
   62|   },
   63|   public: {
   64|     title: 'Public',
   65|     policy: { isPrivate: false, isWebPublic: false, historyPublicToSubscribers: false },
   66|     description:
   67|       'Organization members can join (guests must be invited by a subscriber); organization members can view complete message history without joining',
   68|   },
   69|   private_shared_history: {
   70|     title: 'Private, shared history',
   71|     policy: { isPrivate: true, isWebPublic: false, historyPublicToSubscribers: true },
   72|     description:
   73|       'Must be invited by a subscriber; new subscribers can view complete message history; hidden from non-administrator users',
   74|   },
   75|   private_protected_history: {
   76|     title: 'Private, protected history',
   77|     policy: { isPrivate: true, isWebPublic: false, historyPublicToSubscribers: false },
   78|     description:
   79|       'Must be invited by a subscriber; new subscribers can only see messages sent after they join; hidden from non-administrator users',
   80|   },
   81| };
       ^ [1]
```

So yep, it's being checked.

One way you can test this sort of thing more abstractly is to hover on the values `streamAccessPolicies` and `accessPolicy`, or otherwise ask your editor what Flow thinks the types are. You'll see `AccessPolicy` for the latter, and for the former an object type with those names `web_public` etc. as its four property names. So it does know that those are the object's four properties, and it would give an error if trying to get some arbitrary other property.